Sources/ParseLiveQuery/Internal/QueryEncoder.swift

```diff
@@ -16,14 +16,39 @@ import Parse
 extension Dictionary where Key: StringLiteralConvertible, Value: AnyObject {
     init(query: PFQuery) {
         self.init()
-
+        
         let queryState = query.valueForKey("state")
         if let className = queryState?.valueForKey("parseClassName") {
             self["className"] = className as? Value
         }
-
+        
         if let conditions: [String:AnyObject] = queryState?.valueForKey("conditions") as? [String:AnyObject] {
-            self["where"] = conditions as? Value
+            let newConditions = conditions.sanitizeParseObjects()
+            self["where"] = newConditions as? Value
         }
     }
 }
+
+extension Dictionary {
```

Comment:
We should probably bound the extension with something along the lines of `where Key: String, Value: AnyObject`, which could reduce the number of casts needed in your body of the function.